src/main/java/SolvabilityChecker/MazeSolver.java

```diff
@@ -0,0 +1,66 @@
+package SolvabilityChecker;
```

Comment:
it would make more sense to keep MazeSolver as an entity as it is also used by use case 4 for maze publishing.

if you do this, you could use the getValidMoves function inside HintGenerator.java instead of reimplementing the function.